internal/types/document.go

```diff
@@ -113,27 +127,46 @@ func (d *Document) Len() int {
 	if d == nil {
 		return 0
 	}
-	return len(d.keys)
+
+	return len(d.fields)
 }
 
 // Map returns this document as a map. Do not modify it.
 //
+// If there are duplicate keys in the document, the last value is set in the corresponding field.
+//
 // It returns nil for nil Document.
+//
+// Deprecated: as Document might have duplicate keys, map is not a good representation of it.
```

Comment:
Don't we want to add some function returning a custom struct []{key,value} for that? It would be helpful when you want to access every value inside the document

Reply:
We have other tickets to refactor map-related things. So, probably, yes, but now now :)